cloudigrade/account/tasks.py

```diff
@@ -194,14 +217,57 @@ def run_inspection_cluster(messages, cloud='aws'):
         containerInstances=[result['containerInstanceArns'][0]],
         cluster=settings.HOUNDIGRADE_ECS_CLUSTER_NAME
     )
-    instance_id = result['containerInstances'][0]['ec2InstanceId']
+    ec2_instance_id = result['containerInstances'][0]['ec2InstanceId']
 
-    # attach volumes
+    # Obtain boto EC2 Instance
     ec2 = boto3.resource('ec2')
+    ec2_instance = ec2.Instance(ec2_instance_id)
+
+    logger.info(_('{} attaching volumes').format(
+        'run_inspection_cluster'))
+    # attach volumes
     for index, message in enumerate(messages):
         mount_point = generate_device_name(index)
         volume = ec2.Volume(message['volume_id'])
-        volume.attach_to_instance(Device=mount_point, InstanceId=instance_id)
+        logger.info(_('{} attaching volume {} to instance {}').format(
+            'run_inspection_cluster',
+            message['volume_id'],
+            ec2_instance_id))
+
+        volume.attach_to_instance(
+            Device=mount_point, InstanceId=ec2_instance_id)
+
+        logger.info(_('{} modify volume {} to auto-delete').format(
+            'run_inspection_cluster',
+            message['volume_id']))
+        # Configure volumes to delete when instance is scaled down
+        ec2_instance.modify_attribute(BlockDeviceMappings=[
+            {
+                'DeviceName': mount_point,
+                'Ebs': {
+                    'DeleteOnTermination': True
+                }
+            }
+        ])
+
+        # Remove permissions from customer_snapshot
+        logger.info(_('{} remove ownership from customer snapshot {}').format(
+            'run_inspection_cluster',
+            message['customer_snapshot_id']))
+        session = aws.get_session(message['arn'])
+        customer_snapshot = aws.get_snapshot(
+            session,
+            message['customer_snapshot_id'],
+            message['source_region'])
+        aws.remove_snapshot_ownership(customer_snapshot)
```

Comment:
I _think_ we could move `remove_snapshot_ownership` earlier in the general sequence of tasks to immediately after we start the initial snapshot copy. Once we've started the copy, we may no longer need the permissions, and that means we might not need to pass as much information about the customer (the ARN, the customer's snapshot ID) forward.

Reply:
At the end of copy_ami_snapshot?

Reply:
Nope, too soon there.  :)

Reply:
@infinitewarp We can move it to create_volume function.  The copy snapshot must be complete before we can remove ownership.